cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1774,14 +1774,18 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                                 Secret certSecret = result.result();
 
                                 if (certSecret != null) {
-                                    try {
-                                        X509Certificate cert = Ca.cert(certSecret, customCertSecret.getCertificate());
-                                        byte[] signature = MessageDigest.getInstance("SHA-256").digest(cert.getEncoded());
-                                        thumbprintPromise.complete(Base64.getEncoder().encodeToString(signature));
-                                    } catch (CertificateEncodingException | NoSuchAlgorithmException e) {
-                                        log.warn("{}: Failed to get certificate signature of {} from Secret {}.", reconciliation, customCertSecret.getCertificate(), customCertSecret.getSecretName());
-                                        thumbprintPromise.fail(new RuntimeException("Failed to get certificate signature of " + customCertSecret.getCertificate() + " from Secret " + certSecret.getMetadata().getName(), e));
-                                    }
+                                    if (!certSecret.getData().containsKey(customCertSecret.getCertificate())) {
+                                        thumbprintPromise.fail("Secret " + customCertSecret.getSecretName() + " does not contain certificate " + customCertSecret.getCertificate() + ".");
```

Comment:
Could you add the warn messages as we have them with the other errors to make that consistent? Also, I think the message could you some more clarification ... maybe something like _Secret xyz does not contian certificate under the key <tls.crt>_.